Makefile

```diff
@@ -29,6 +34,50 @@ test:
 	@echo "running against `git version`"; \
 	$(GOTEST) -race ./...
 
+TEST_ENV=.env
+DOCKER_ENV=.env-docker
+DOCKER_GOGIT_KEY = go-git
+DOCKER_GOGIT_LABEL = testing
+DOCKER_GOGIT_NAME = $(DOCKER_GOGIT_KEY)-dist
+.PHONY: test-env test-go1.19 test-go1.20 test-go1.21 test-go-all
+test-env:
+	cat <<EOF >$(TEST_ENV)
+	DOCKER_ENV=$(DOCKER_ENV)
+	DOCKER_GOGIT_KEY=$(DOCKER_GOGIT_KEY)
+	DOCKER_GOGIT_LABEL=$(DOCKER_GOGIT_LABEL)
+	DOCKER_GOGIT_NAME=$(DOCKER_GOGIT_NAME)
+	GIT_REPOSITORY=$(GIT_REPOSITORY)
+	GIT_VERSIONS=(master v2.11.0)
+	GO_VERSION_WGIT=1.21
+	GO_VERSIONS=(1.19 1.20 1.21)
+	WORKDIR=$(WORKDIR)
+	EOF
+
+test-go1.19: test-env
+	export BASH_ENV=$(TEST_ENV)
+	$(BASHCMD) _local/test-go.sh "1.19"
+
+test-go1.20: test-env
+	export BASH_ENV=$(TEST_ENV)
+	$(BASHCMD) _local/test-go.sh "1.20"
+
+test-go1.21: test-env
+	BASH_ENV=$(TEST_ENV)
+	$(BASHCMD) _local/test-go.sh "1.21"
```

Comment:
If we have target per Go version, we will need to update this twice a year. What about we have a single Makefile target and grab the Go version via environment var?

Reply:
So you would rather have one target, which possibly never needs to change and it would be run something like this?

```shell
GO_VERSION=1.21; make test-go
```